tests/lib/rules/no-extra-parens.js

```diff
@@ -291,7 +314,61 @@ ruleTester.run("no-extra-parens", rule, {
         { code: "foo instanceof (bar instanceof baz)", options: ["all", { nestedBinaryExpressions: false }] },
         { code: "foo in (bar in baz)", options: ["all", { nestedBinaryExpressions: false }] },
         { code: "foo + (bar + baz)", options: ["all", { nestedBinaryExpressions: false }] },
-        { code: "foo && (bar && baz)", options: ["all", { nestedBinaryExpressions: false }] }
+        { code: "foo && (bar && baz)", options: ["all", { nestedBinaryExpressions: false }] },
+
+        // ignoreJSX === "all"
+        jsx("const Component = (<div />)", ["all", { ignoreJSX: "all" }]),
```

Comment:
In my opinion, this call signature is a bit confusing -- I had no idea what was going on until I looked at the function signature above.

I would recommend either:

* Use defaults in the `RuleTester` constructor
```js
const ruleTester = new RuleTester({
    parserOptions: { ecmaVersion: 6, ecmaFeatures: { jsx: true } }
});
```

* Make the `jsx` function accept an object, instead of two parameters. This would make the jsx test cases look similar to everything else:

```js
    {
        code: "foo",
        options: ["bar"]
    },
    jsx({
        code: "foo",
        options: ["baz"]
    }),
```